src/coreclr/jit/compiler.cpp

```diff
@@ -5157,12 +5157,22 @@ void Compiler::compCompile(void** methodCodePtr, uint32_t* methodCodeSize, JitFl
         m_llvm->Lower();
     });
 
-    fgSsaDomTree = nullptr;
     if (opts.OptimizationEnabled())
     {
+        DoPhase(this, PHASE_DFS_BLOCKS, &Compiler::fgDfsBlocksAndRemove);
+
+        // Clear the immediate dominators so that we process the scratch block.
+        for (BasicBlock* const block : Blocks())
+        {
+            block->bbIDom         = nullptr;
+            block->bbPostorderNum = 0;
+        }
```

Comment:
This looks like a bit of an upstream issue that this needs to be done explicitly. It is currently hidden in the OptRepeat case by the fact legacy `fgComputeDoms` clears the immediate dominators as part of `RecomputeFlowGraphAnnotations -> fgComputeReachability`, but it'll surface one that's removed.

It is fine to leave it be for the time being, however.
```suggestion
        // Clear immediate dominators left over from the optimization phases.
        for (BasicBlock* const block : Blocks())
        {
            block->bbIDom         = nullptr;
            block->bbPostorderNum = 0;
        }
```
BTW, I wonder if you need to clear `block->bbPostorderNum` here. It is the legacy field, theoretically not used by anything downstream.

Reply:
you are right I can remove `bbPostorderNum`, thanks.

Reply:
I wouldn't expect this to be necessary. What issues were you hitting without it?

Reply:
I think this is answered in the discord by SA.  Let me know if you want more details, but it seems like different solution has already been proposed and there is now an issue #2478